### ISSUE TITLE
docker zap-api-scan returns "Number of Imported URLs: 0"

### ISSUE THREAD

**Issue by leoweiyu** [zaproxy/zaproxy#5428]
**Describe the bug**
we manged to run ZAP on OpenShift, the zap-full-scan and zap-baseline works as expected, but whenever we tried to run following Jenkins pipeline to trigger an API scan
`stage('Get a ZAP Pod') {
    node('zap') {
        stage('Scan Web Application') {
            //sh 'mkdir /tmp/workdir'
            dir('/tmp/workdir') {
                sh (
                  script: '''
                    curl http://whatever11:8651/TIP/gaga.json -o /zap/wrk/gaga.json
                  ''',
                  returnStdout: true
                ).trim()
                def retVal = sh returnStatus: true, script: '/zap/zap-api-scan.py -d -r baseline.html -t gaga.json -f openapi'
                publishHTML([allowMissing: false, alwaysLinkToLastBuild: false, keepAll: true, reportDir: '/zap/wrk', reportFiles: 'baseline.html', reportName: 'ZAP Baseline Scan', reportTitles: 'ZAP Baseline Scan'])
                echo "Return value is: ${retVal}"
            }
        }
    }`

it always return 
```
2019-06-19 16:03:50,356 Number of Imported URLs: 0
2019-06-19 16:03:50,356 Import warnings: No Implementor
2019-06-19 16:03:50,356 Failed to import any URLs
Traceback (most recent call last):
  File "/zap/zap-api-scan.py", line 397, in main
    raise NoUrlsException()
```

we have tried both WSDL and OpenAPI files, none of them working which return us exactly the same error message
i have attached both wsdl file and OpenAPI files
[w3s.txt](https://github.com/zaproxy/zaproxy/files/3305187/w3s.txt)

[swagger.txt](https://github.com/zaproxy/zaproxy/files/3305190/swagger.txt)

**To Reproduce**
1. run zap docker image as a jenkins slave
2. bring up he slave in pipeline shown above
3. the slave will then call /zap/zap-api-scan.py -d -r baseline.html -t swagger.json -f openapi
or /zap/zap-api-scan.py -d -r baseline.html -t w3s.xml -f soap

**Expected behavior**
the wsdl or openapi file get parsed correctly and zap will then trigger scanning against URL found in the wsdl or openapi files

**Screenshots**
N/A

**Software versions**
 - ZAP: lasted version (develop Version)
 - Add-on: using the docker file provided 
 - OS: using the docker file provided 
 - Java: using the docker file provided 
 - Browser: N/A

**Errors from the zap.log file**

> Still waiting to schedule task
> ‘jenkins-slave-m7fjh-n5gzn’ is offline
> Agent jenkins-slave-m7fjh-n5gzn is provisioned from template Kubernetes Pod Template
> Agent specification [Kubernetes Pod Template] (zap): 
> * [jnlp] docker-registry.default.svc:5000/test/test-zap(resourceRequestCpu: 500m, resourceRequestMemory: 1Gi, resourceLimitCpu: 1, resourceLimitMemory: 2Gi)
> 
> Running on jenkins-slave-m7fjh-n5gzn in /tmp/workspace/test-dev/test-dev-zap-demo
> [Pipeline] {
> [Pipeline] stage
> [Pipeline] { (Scan Web Application)
> [Pipeline] dir
> Running in /tmp/workdir
> [Pipeline] {
> [Pipeline] sh
> + curl http://drlfic11:8651/TIP/gaga.json -o /zap/wrk/swagger.json
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
> 
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
> 100 24949  100 24949    0     0   322k      0 --:--:-- --:--:-- --:--:--  324k
> [Pipeline] sh
> + /zap/zap-api-scan.py -d -r baseline.html -t gaga.json -f openapi
> 2019-06-19 16:02:15,989 Target: gaga.json
> 2019-06-19 16:02:15,989 Could not find custom hooks file at /var/lib/jenkins/.zap_hooks.py 
> 2019-06-19 16:02:15,989 Trigger hook: cli_opts, args: 1
> 2019-06-19 16:02:15,990 Using port: 60706
> 2019-06-19 16:02:15,990 Trigger hook: start_zap, args: 2
> 2019-06-19 16:02:15,990 Starting ZAP
> 2019-06-19 16:02:15,990 Params: ['zap-x.sh', '-daemon', '-port', '60706', '-host', '0.0.0.0', '-config', 'api.disablekey=true', '-config', 'api.addrs.addr.name=.*', '-config', 'api.addrs.addr.regex=true', '-addonupdate', '-addoninstall', 'pscanrulesBeta']
> 2019-06-19 16:02:15,995 Starting new HTTP connection (1): localhost:60706
> _XSERVTransmkdir: ERROR: euid != 0,directory /tmp/.X11-unix will not be created.
> OpenJDK 64-Bit Server VM warning: If the number of processors is expected to increase from one, then you should configure the number of parallel GC threads appropriately using -XX:ParallelGCThreads=N
> 2019-06-19 16:02:16,998 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:18,001 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:19,004 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:20,006 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:21,009 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:22,012 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:23,015 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:24,017 Starting new HTTP connection (1): localhost:60706
> Jun 19, 2019 4:02:24 PM java.util.prefs.FileSystemPreferences$1 run
> INFO: Created user preferences directory.
> 2019-06-19 16:02:25,020 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:26,023 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:27,025 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:28,028 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:29,032 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:30,036 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:31,039 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:32,042 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:33,045 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:34,048 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:35,051 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:36,054 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:37,058 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:38,060 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:39,063 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:40,066 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:41,069 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:42,072 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:43,076 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:44,081 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:45,084 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:46,087 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:47,090 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:48,094 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:49,096 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:50,100 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:51,103 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:52,105 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:53,109 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:54,111 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:55,114 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:56,116 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:57,119 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:58,123 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:02:59,125 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:00,128 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:01,131 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:02,134 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:03,137 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:04,139 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:05,142 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:06,145 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:07,148 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:08,151 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:09,154 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:10,157 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:11,159 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:12,162 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:13,165 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:14,169 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:15,172 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:16,175 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:17,178 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:18,182 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:19,185 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:20,188 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:21,191 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:22,194 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:23,197 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:24,200 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:25,203 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:26,206 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:27,209 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:28,212 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:29,215 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:30,218 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:31,222 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:32,225 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:33,228 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:34,230 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:35,233 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:36,237 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:37,240 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:38,243 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:39,246 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:40,249 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:41,252 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:42,255 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:43,258 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:44,261 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:45,264 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:46,267 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:47,270 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:48,273 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:49,276 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:50,279 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:50,325 http://localhost:60706 "GET http://zap/JSON/core/view/version/ HTTP/1.1" 200 19
> 2019-06-19 16:03:50,326 ZAP Version 2.8.0
> 2019-06-19 16:03:50,326 Took 94 seconds
> 2019-06-19 16:03:50,326 Trigger hook: zap_started, args: 2
> 2019-06-19 16:03:50,327 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:50,332 http://localhost:60706 "GET http://zap/JSON/script/action/load/?scriptEngine=Oracle+Nashorn&scriptName=Alert_on_HTTP_Response_Code_Errors.js&scriptType=httpsender&apikey=&fileName=%2Fhome%2Fzap%2F.ZAP_D%2Fscripts%2Fscripts%2Fhttpsender%2FAlert_on_HTTP_Response_Code_Errors.js HTTP/1.1" 400 52
> 2019-06-19 16:03:50,333 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:50,336 http://localhost:60706 "GET http://zap/JSON/script/action/enable/?scriptName=Alert_on_HTTP_Response_Code_Errors.js&apikey= HTTP/1.1" 400 52
> 2019-06-19 16:03:50,338 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:50,340 http://localhost:60706 "GET http://zap/JSON/script/action/load/?scriptEngine=Oracle+Nashorn&scriptName=Alert_on_Unexpected_Content_Types.js&scriptType=httpsender&apikey=&fileName=%2Fhome%2Fzap%2F.ZAP_D%2Fscripts%2Fscripts%2Fhttpsender%2FAlert_on_Unexpected_Content_Types.js HTTP/1.1" 400 52
> 2019-06-19 16:03:50,342 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:50,345 http://localhost:60706 "GET http://zap/JSON/script/action/enable/?scriptName=Alert_on_Unexpected_Content_Types.js&apikey= HTTP/1.1" 400 52
> 2019-06-19 16:03:50,346 Trigger hook: importing_openapi, args: 2
> 2019-06-19 16:03:50,346 Import OpenAPI File gaga.json
> 2019-06-19 16:03:50,347 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:50,349 http://localhost:60706 "GET http://zap/JSON/openapi/action/importFile/?apikey=&file=%2Fzap%2Fwrk%2Fgaga.json HTTP/1.1" 400 52
> 2019-06-19 16:03:50,350 Starting new HTTP connection (1): localhost:60706
> 2019-06-19 16:03:50,355 http://localhost:60706 "GET http://zap/JSON/core/view/urls/ HTTP/1.1" 200 11
> 2019-06-19 16:03:50,356 Number of Imported URLs: 0
> 2019-06-19 16:03:50,356 Import warnings: No Implementor
> 2019-06-19 16:03:50,356 Failed to import any URLs
> Traceback (most recent call last):
>   File "/zap/zap-api-scan.py", line 397, in main
>     raise NoUrlsException()
> NoUrlsException
> 2019-06-19 16:03:50,356 Failed to find zap_log /zap/zap.out
> 2019-06-19 16:03:50,356 Trigger hook: pre_exit, args: 3
> [Pipeline] publishHTML
> [htmlpublisher] Archiving HTML reports...
> [htmlpublisher] Archiving at BUILD level /zap/wrk to /var/lib/jenkins/jobs/test-dev/jobs/test-dev-zap-demo/builds/25/htmlreports/ZAP_20Baseline_20Scan
> [Pipeline] echo
> Return value is: 3

**Additional context**

**Comment by thc202** [comment]
Based on the output provided the import of the API definition failed, you would have to check the ZAP log (it should be shown in the output of the script but `Failed to find zap_log /zap/zap.out`).

**Comment by leoweiyu** [comment]
thanks @thc202 , i have found following two exceptions in zap.out

> 48239 [ZAP-daemon] ERROR org.parosproxy.paros.CommandLine  - Check for updates call failed
> 88305 [ZAP-cfu] WARN org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate  - Failed to check for updates using: https://raw.githubusercontent.com/zaproxy/zap-admin/master/ZapVersions-2.8.xml
> java.net.SocketTimeoutException: connect timed out
>         at java.net.PlainSocketImpl.socketConnect(Native Method)
>         at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
>         at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
>         at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
>         at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
>         at java.net.Socket.connect(Socket.java:589)
>         at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:666)
>         at org.parosproxy.paros.network.SSLConnector.createSocket(SSLConnector.java:450)
>         at org.apache.commons.httpclient.HttpConnection.open(HttpConnection.java:728)
>         at org.apache.commons.httpclient.MultiThreadedHttpConnectionManager$HttpConnectionAdapter.open(MultiThreadedHttpConnectionManager.java:1361)
>         at org.apache.commons.httpclient.HttpMethodDirector.executeWithRetry(HttpMethodDirector.java:449)
>         at org.apache.commons.httpclient.HttpMethodDirector.executeMethod(HttpMethodDirector.java:201)
>         at org.apache.commons.httpclient.HttpClient.executeMethod(HttpClient.java:397)
>         at org.parosproxy.paros.network.HttpSender.executeMethod(HttpSender.java:398)
>         at org.parosproxy.paros.network.HttpSender.runMethod(HttpSender.java:633)
>         at org.parosproxy.paros.network.HttpSender.send(HttpSender.java:589)
>         at org.parosproxy.paros.network.HttpSender.sendAuthenticated(HttpSender.java:565)
>         at org.parosproxy.paros.network.HttpSender.sendAuthenticated(HttpSender.java:553)
>         at org.parosproxy.paros.network.HttpSender.sendAndReceive(HttpSender.java:458)
>         at org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate.getRemoteConfigurationUrl(ExtensionAutoUpdate.java:1046)
>         at org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate.access$900(ExtensionAutoUpdate.java:92)
>         at org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate$8.run(ExtensionAutoUpdate.java:1211)
> 89247 [ZAP-daemon] ERROR org.parosproxy.paros.CommandLine  - Check for updates call failed
> 89248 [ZAP-daemon] INFO org.zaproxy.zap.DaemonBootstrap  - ZAP is now listening on 0.0.0.0:41121
> 89346 [ZAP-ProxyThread-6] WARN org.zaproxy.zap.extension.api.API  - Bad request to API endpoint [/JSON/soap/action/importFile/] from [0:0:0:0:0:0:0:1]:
> No Implementor (no_implementor)
>         at org.zaproxy.zap.extension.api.API.handleApiRequest(API.java:433)
>         at org.parosproxy.paros.core.proxy.ProxyThread.processHttp(ProxyThread.java:499)
>         at org.parosproxy.paros.core.proxy.ProxyThread.run(ProxyThread.java:335)
>         at java.lang.Thread.run(Thread.java:748)


i kinda understand why we getting the first exception which failed to grab the update file as we are behind corporate proxy, is there a way we can specify proxy settting by using zap-api-scan.py?

and the second exception is the root cause of our problem, which unable to parse the WSDL file passed to the zap-api-scan.py, the zap.out simply say its a bad request to API endpoint /JSON/soap/action/importFile, but i have no idea whyit is a bad request, could you please shed some light on that please

i have the full log attached

[zap.txt](https://github.com/zaproxy/zaproxy/files/3308227/zap.txt)


and my jenkins log shows following

> 2019-06-20 11:33:19,419 ZAP Version 2.8.0
> 2019-06-20 11:33:19,419 Took 91 seconds
> 2019-06-20 11:33:19,419 Trigger hook: zap_started, args: 2
> 2019-06-20 11:33:19,420 Starting new HTTP connection (1): localhost:41121
> 2019-06-20 11:33:19,425 http://localhost:41121 "GET http://zap/JSON/script/action/load/?scriptEngine=Oracle+Nashorn&scriptName=Alert_on_HTTP_Response_Code_Errors.js&scriptType=httpsender&apikey=&fileName=%2Fhome%2Fzap%2F.ZAP_D%2Fscripts%2Fscripts%2Fhttpsender%2FAlert_on_HTTP_Response_Code_Errors.js HTTP/1.1" 200 15
> 2019-06-20 11:33:19,426 Starting new HTTP connection (1): localhost:41121
> 2019-06-20 11:33:19,429 http://localhost:41121 "GET http://zap/JSON/script/action/enable/?scriptName=Alert_on_HTTP_Response_Code_Errors.js&apikey= HTTP/1.1" 200 15
> 2019-06-20 11:33:19,430 Starting new HTTP connection (1): localhost:41121
> 2019-06-20 11:33:19,432 http://localhost:41121 "GET http://zap/JSON/script/action/load/?scriptEngine=Oracle+Nashorn&scriptName=Alert_on_Unexpected_Content_Types.js&scriptType=httpsender&apikey=&fileName=%2Fhome%2Fzap%2F.ZAP_D%2Fscripts%2Fscripts%2Fhttpsender%2FAlert_on_Unexpected_Content_Types.js HTTP/1.1" 200 15
> 2019-06-20 11:33:19,434 Starting new HTTP connection (1): localhost:41121
> 2019-06-20 11:33:19,436 http://localhost:41121 "GET http://zap/JSON/script/action/enable/?scriptName=Alert_on_Unexpected_Content_Types.js&apikey= HTTP/1.1" 200 15
> 2019-06-20 11:33:19,436 Trigger hook: importing_soap, args: 2
> 2019-06-20 11:33:19,436 Import SOAP File gaga.xml
> 2019-06-20 11:33:19,437 Starting new HTTP connection (1): localhost:41121
> 2019-06-20 11:33:19,440 http://localhost:41121 "GET http://zap/JSON/soap/action/importFile/?file=%2Fzap%2Fwrk%2Fgaga.xml HTTP/1.1" 400 52
> 2019-06-20 11:33:19,441 Starting new HTTP connection (1): localhost:41121
> 2019-06-20 11:33:19,445 http://localhost:41121 "GET http://zap/JSON/core/view/urls/ HTTP/1.1" 200 11
> 2019-06-20 11:33:19,445 Number of Imported URLs: 0
> 2019-06-20 11:33:19,445 Import warnings: {u'message': u'No Implementor', u'code': u'no_implementor'}
> 2019-06-20 11:33:19,445 Failed to import any URLs
> Traceback (most recent call last):
>   File "/zap/zap-api-scan.py", line 397, in main
>     raise NoUrlsException()
> NoUrlsException
> 2019-06-20 11:33:19,446 Failed to find zap_log /zap/zap.out
> 2019-06-20 11:33:19,446 Trigger hook: pre_exit, args: 3

kind regards

Leo Yu

**Comment by kingthorin** [comment]
The soap API endpoints only exist if the soap extension is installed. According to the log file you attached it isn't available:
```
4149 [ZAP-daemon] INFO org.zaproxy.zap.control.ExtensionFactory  - Installed add-ons: [[id=alertFilters, version=8.0.0], [id=ascanrules, version=33.0.0], [id=bruteforce, version=8.0.0], [id=diff, version=9.0.0], [id=directorylistv1, version=3.0.0], [id=fuzz, version=11.0.0], [id=gettingStarted, version=10.0.0], [id=help, version=9.0.0], [id=hud, version=0.4.0], [id=importurls, version=6.0.0], [id=invoke, version=9.0.0], [id=jxbrowser, version=14.0.0], [id=jxbrowserlinux64, version=12.0.0], [id=onlineMenu, version=6.0.0], [id=pscanrules, version=24.0.0], [id=quickstart, version=26.0.0], [id=replacer, version=7.0.0], [id=reveal, version=2.0.0], [id=saverawmessage, version=4.0.0], [id=savexmlmessage, version=0.0.1], [id=scripts, version=25.0.0], [id=selenium, version=15.0.0], [id=spiderAjax, version=23.0.0], [id=tips, version=6.0.0], [id=webdriverlinux, version=10.0.0], [id=websocket, version=19.0.0], [id=zest, version=29.0.0]]
```

`soap/action/importFile`
`soap/action/importUrl`

Due to outstanding issues the SOAP Extension was de-published last summer: https://github.com/zaproxy/zaproxy/issues/4866

**Comment by leoweiyu** [comment]
sweet, thanks @kingthorin and @thc202  it worked after i chucked openapi and soap plugin, thanks heaps, but by any chance you could advise how i can configure proxy setting for zaproxy, we running the docker image behind corporate proxy, thats why  it failed to update while starting up

> Failed to check for updates using: https://raw.githubusercontent.com/zaproxy/zap-admin/master/ZapVersions-2.8.xml

i have tried to set both HTTP_PROXY and HTTPS_PROXY as environment variable in the container, but seems zap run as a java process which doesn't pick those thing up, is there a way i can set -Dhttps.proxyHost for zaproxy?

kind regards

Leo

**Comment by kingthorin** [comment]
> but by any chance you could advise how i can configure proxy setting for zaproxy, we running the docker image behind corporate proxy, thats why it failed to update while starting up

@leoweiyu just set zap’s outbound proxy

https://github.com/zaproxy/zap-core-help/wiki/HelpUiDialogsOptionsConnection#use-proxy-chain

**Comment by zecarrera** [comment]
> The soap API endpoints only exist if the soap extension is installed. According to the log file you attached it isn't available:
> 
> ```
> 4149 [ZAP-daemon] INFO org.zaproxy.zap.control.ExtensionFactory  - Installed add-ons: [[id=alertFilters, version=8.0.0], [id=ascanrules, version=33.0.0], [id=bruteforce, version=8.0.0], [id=diff, version=9.0.0], [id=directorylistv1, version=3.0.0], [id=fuzz, version=11.0.0], [id=gettingStarted, version=10.0.0], [id=help, version=9.0.0], [id=hud, version=0.4.0], [id=importurls, version=6.0.0], [id=invoke, version=9.0.0], [id=jxbrowser, version=14.0.0], [id=jxbrowserlinux64, version=12.0.0], [id=onlineMenu, version=6.0.0], [id=pscanrules, version=24.0.0], [id=quickstart, version=26.0.0], [id=replacer, version=7.0.0], [id=reveal, version=2.0.0], [id=saverawmessage, version=4.0.0], [id=savexmlmessage, version=0.0.1], [id=scripts, version=25.0.0], [id=selenium, version=15.0.0], [id=spiderAjax, version=23.0.0], [id=tips, version=6.0.0], [id=webdriverlinux, version=10.0.0], [id=websocket, version=19.0.0], [id=zest, version=29.0.0]]
> ```
> 
> `soap/action/importFile`
> `soap/action/importUrl`
> 
> Due to outstanding issues the SOAP Extension was de-published last summer: #4866
> 
> (You can find the previously published version here: https://github.com/zaproxy/zap-extensions/releases/tag/2.7, however, it will be subject to the issues outlined in #4866)

Thanks.. That worked as well for me. 
In my case, I'm running it in docker using the zap2docker-stable, and all I needed to do was change my dockerfile to get the extenstion.
RUN wget -q "https://github.com/zaproxy/zap-extensions/releases/download/2.7/openapi-alpha-12.zap" -O /zap/plugin/openapi-alpha-12.zap
